packages/package-manager/src/utils/__tests__/shouldUseYarn-test.ts

```diff
@@ -11,6 +11,13 @@ beforeAll(() => {
       },
     };
   });
+  jest.mock('../nodeWorkspaces', () => {
+    return {
+      isUsingNpm: () => {
+        return !!process.env.TEST_NPM_LOCKFILE_EXISTS;
+      },
+    };
+  });
 });
```

Comment:
```suggestion
  jest.mock('../nodeWorkspaces', () => ({ isUsingNpm: jest.fn() }))
});
```
Then import `isUsingNpm` and do something like this:



```ts
import { isUsingNpm } from '../path/to/thing';

const asMock = <T extends (...args: any[]) => any>(fn: T): jest.MockedFunction<T> =>
  fn as jest.MockedFunction<T>;

it(`does foobar`, () => {
  // Mock what should be returned for the test case.
  asMock(isUsingNpm).mockClear().mockReturnValueOnce(true);
})


```